crates/noirc_evaluator/src/ssa/code_gen.rs

```diff
@@ -95,6 +95,20 @@ impl Value {
             Value::Tuple(fields) => &fields[field_index],
         }
     }
+
+    //Deflate a vector of NodeIds into a Value whose type is provided in argument
+    pub fn deflate(value_type: &Type, iter: &mut core::slice::Iter<NodeId>) -> Value {
```

Comment:
This mostly looks good to me, though I have 2 suggestions:
1. `deflate` sounds to me like it is doing the opposite, can we rename this to a clearer `from_array`?
2. I think this function should be a private function (it is unused outside this file) and we should have a wrapper function over it accepting a slice directly instead of an iterator. This way when the wrapper is done calling the helper we can verify the helper actually finished consuming the entire slice and that there are no remaining elements we missed.

Reply:
I agree with the first point and that the function can be made private. Deflate seems like it would be flattening the Value while it is doing the opposite. I would propose the term `reshape` or `reverse_flatten`?

Reply:
`reverse_flatten` sounds like `foo.flatten().reverse()` to me. I'd prefer `unflatten` or `from_array`

Reply:
Yeah good point `reverse_flatten` would be confusing. So let's not do that one. 

I like both `unflatten` and `from_slice`. I still like `reshape` as well, but I think any of these names are clear and would be fine with any of them.

Reply:
Thank you for the comments and sorry for my wrong interpretation of deflate. Thinking about it I remember it also fooled me when I added compression to the circuit file.
I have renamed the function.